build.sbt

```diff
@@ -66,77 +67,47 @@ val manual =
       addMappingsToSiteDir(mappings in ScalaUnidoc in packageDoc in apiDoc, siteSubdirName in packageDoc)
     )
 
-import ReleaseTransformations._
-
 val `scalajs-bundler` =
   project.in(file("."))
     .settings(noPublishSettings: _*)
-    .settings(
-      releaseProcess := Seq[ReleaseStep](
-        checkSnapshotDependencies,
-        inquireVersions,
-        runClean,
-        runTest,
-        releaseStepInputTask(scripted in `sbt-scalajs-bundler`),
-        releaseStepInputTask(scripted in `sbt-web-scalajs-bundler`),
-        releaseStepTask(ornate in manual),
-        setReleaseVersion,
-        commitReleaseVersion,
-        tagRelease,
-        releaseStepTask(PgpKeys.publishSigned in `sbt-scalajs-bundler`),
-        releaseStepTask(PgpKeys.publishSigned in `sbt-web-scalajs-bundler`),
-        setNextVersion,
-        commitNextVersion,
-        pushChanges,
-        releaseStepTask(GhPagesKeys.pushSite in manual)
-      )
-    )
-    .aggregate(`sbt-scalajs-bundler`, `sbt-web-scalajs-bundler`, manual, apiDoc)
+    .aggregate(`sbt-scalajs-bundler`, `sbt-web-scalajs-bundler`)
 
-lazy val commonSettings =
-  ScriptedPlugin.scriptedSettings ++ Seq(
-    scalacOptions ++= Seq(
-      "-feature",
-      "-deprecation",
-      "-encoding", "UTF-8",
-      "-unchecked",
-      "-Xlint",
-      "-Yno-adapted-args",
-      "-Ywarn-dead-code",
-      "-Ywarn-numeric-widen",
-      "-Ywarn-value-discard",
-      "-Xfuture"
-    ),
-    organization := "ch.epfl.scala",
-    pomExtra :=
-      <developers>
-        <developer>
-          <id>julienrf</id>
-          <name>Julien Richard-Foy</name>
-          <url>http://julien.richard-foy.fr</url>
-        </developer>
-      </developers>,
-    homepage := Some(url(s"https://github.com/scalacenter/scalajs-bundler")),
-    licenses := Seq("MIT License" -> url("http://opensource.org/licenses/mit-license.php")),
-    scmInfo := Some(
-      ScmInfo(
-        url("https://github.com/scalacenter/scalajs-bundler"),
-        "scm:git:git@github.com:scalacenter/scalajs-bundler.git"
-      )
-    ),
-    crossSbtVersions := List("0.13.16", "1.0.2"),
-    scalaVersion := {
-      (sbtBinaryVersion in pluginCrossBuild).value match {
-        case "0.13" => "2.10.6"
-        case _ => "2.12.3"
-      }
-    },
-    // fixed in https://github.com/sbt/sbt/pull/3397 (for sbt 0.13.17)
-    sbtBinaryVersion in update := (sbtBinaryVersion in pluginCrossBuild).value,
-    runScripted := runScriptedTask.value,
-    scriptedLaunchOpts += "-Dplugin.version=" + version.value,
-    scriptedBufferLog := false
-  )
+inThisBuild(List(
+  pgpPublicRing := file("./travis/local.pubring.asc"),
+  pgpSecretRing := file("./travis/local.secring.asc"),
+  releaseEarlyWith := SonatypePublisher,
+  scalacOptions ++= Seq(
```

Comment:
Are these being used at all? I think they're ignored by all the projects.

Reply:
Why are they ignored? My goal is that both `sbt-scalajs-bundler` and `sbt-web-scalajs-bundler` use these options.

Reply:
They're ignored because `scalacOptions` is defined in the configurations of the projects, not the projects. Therefore, sbt will never delegate to this one. This is confirmed by the "Provided by" section:
  
```scala
sbt @ caca > inspect scalacOptions
[info] Task: scala.collection.Seq[java.lang.String]
[info] Description:
[info] 	Options for the Scala compiler.
[info] Provided by:
[info] 	{file:/data/rw/code/scala/caca/}caca/compile:scalacOptions
[info] Defined at:
[info] 	(sbt.Classpaths) Defaults.scala:1871
[info] 	(sbt.Defaults) Defaults.scala:1041
[info] Dependencies:
[info] 	Compile/sbtPlugin
[info] 	Compile/settingsData
[info] 	Compile/buildDependencies
[info] 	Compile/update
[info] 	Compile/autoCompilerPlugins
[info] 	Compile/thisProjectRef
[info] Reverse dependencies:
[info] 	Compile/ensimeCompileOnly/scalacOptions
[info] 	ensimeScalacOptions
[info] 	Compile/argsFileContents
[info] Delegates:
[info] 	Compile/scalacOptions
[info] 	scalacOptions
[info] 	ThisBuild/Compile/scalacOptions
[info] 	ThisBuild/scalacOptions
[info] 	Zero/Compile/scalacOptions
[info] 	Global/scalacOptions
[info] Related:
[info] 	Test/ensimeCompileOnly/scalacOptions
[info] 	Compile/ensimeCompileOnly/scalacOptions
[info] 	Global/scalacOptions
[info] 	Test/scalacOptions
```

Reply:
That's not true. Because the original definition of  `scalacOptions in (proj, Compile)` is copied from `scalacOptions in proj`, which itself is copied from `scalacOptions in ThisBuild`.

Those settings are not ignored.

Reply:
Yes, you're right.